core/src/main/java/org/apache/carbondata/core/metadata/converter/ThriftWrapperSchemaConverterImpl.java

```diff
@@ -605,6 +605,14 @@ public TableSchema fromExternalToWrapperTableSchema(
           .setLocalDictColumnsToWrapperSchema(listOfColumns, externalTableSchema.tableProperties,
               externalTableSchema.tableProperties
                   .get(CarbonCommonConstants.LOCAL_DICTIONARY_ENABLE));
+      String handler = externalTableSchema.tableProperties.get(CarbonCommonConstants.INDEX_HANDLER);
+      if (handler != null) {
+        for (ColumnSchema columnSchema : listOfColumns) {
+          if (handler.equalsIgnoreCase(columnSchema.getColumnName())) {
```

Comment:
Why are you checking column name against handler? 
Either of these is not a proper name

Reply:
index column was created with index_handler property value as its name during create table in `processIndexProperty` method. So matching handler value against the column name.

Reply:
I strongly agree with @jackylk, we already have many index like SI, bloom, lucene. 
This index handler and isIndex column is confusing. 
so we should have something like `spatialIndexHandler` and `isSpatialIndexColumn` as we use these interface for geo spatial / spatial only now.

But renaming thrift will again create compatibility issue if somebody started geospatial already.

@VenuReddy2103 : we can have that renaming change in another PR.

Reply:
@ajantha-bhat I think this feature is still alpha feature in 2.0, 1.6 does not include this feature. For long term benefit, I still think these names should be changed to make it clear to user and developers